File: Rearrange Array Elements by Sign.py
class Solution(object):
    def rearrangeArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        arr = [0] * len(nums)  
        pos_index, neg_index = 0, 1  

        for num in nums:
            if num > 0:
                arr[pos_index] = num
                pos_index += 2 
            else:
                arr[neg_index] = num
                neg_index += 2  

        return arr

File: Rearrange Array Elements by Sign.cpp
class Solution {
    public:
        vector<int> rearrangeArray(vector<int>& nums) {
            vector<int>pos;
            vector<int>neg;
            for (auto num: nums)
            {
                if(num>0)
                {
                    pos.push_back(num);
                }
                else
                {
                    neg.push_back(num);
                }
            }
            int i=0, j=0, k=0;
            vector<int>v;
            while(i<nums.size())
            {
                if(i%2==0)
                {
                    v.push_back(pos[j++]);
                }
                else
                {
                    v.push_back(neg[k++]);
                }
                i++;
            }
            return v;
        }
    };